### PR TITLE
Add android udev rules package

### PR DESCRIPTION
**Pull request by rien**
This pr adds the [android-udev-rules](https://github.com/M0Rf30/android-udev-rules) package, an udev rules list for android devices.

If one installs the `android-tools` (the `adb` command), they will almost always want this installed too. So I've added this package as a dependency.

### CODE REVIEW

File: srcpkgs/android-udev-rules/template [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# Template file for 'android-udev-rules'
+pkgname=android-udev-rules
+version=20200613
+revision=1
+build_style=fetch
```
Comment (sgn) [inline srcpkgs/android-udev-rules/template:5]: - Remove `build_style`.

```sh
do_install() {
    vinstall 51-android.rules" 644 usr/lib/udev/rules.d
}
```

Commit message should be: `New package: android-udev-rules-20200613`

File: srcpkgs/android-udev-rules/template [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+short_desc="Android udev rules list aimed to be the most comprehensive on the net"
+maintainer="Rien Maertens <rien.maertens@posteo.be>"
+license="GPL-3.0-or-later"
+homepage="https://github.com/M0Rf30/${pkgname}"
+distfiles="https://github.com/M0Rf30/${pkgname}/archive/${version}.tar.gz"
```
Comment (sgn) [inline srcpkgs/android-udev-rules/template:10]: don't use `${pkgname}` here, write it in full.

> srcpkgs/android-udev-rules/template:9: homepage should not use variables

File: srcpkgs/android-tools/template [hunk 3]
```diff
@@ -6,6 +6,7 @@ archs="armv* aarch64* x86_64* i686*"
 build_style=cmake
 hostmakedepends="perl go"
 makedepends="gtest-devel zlib-devel libressl-devel libusb-devel pcre2-devel"
+depends="android-udev-rules"
```
Comment (sgn) [inline srcpkgs/android-tools/template:9]: bump `revision`
commit message: `android-tools: depends on android-udev-tools`

**Comment by Duncaen** [comment]
This adds a group to the devices that doesn't exist.

**Comment by Duncaen** [comment]
Also not a fan of depending on this by default if android-tools is installed.

**Comment by sgn** [comment]
Sorry, I have `adbusers` group in my setup.
```diff
diff --git a/srcpkgs/android-udev-rules/template b/srcpkgs/android-udev-rules/template
index 83d9636a6f..9fc589f713 100644
--- a/srcpkgs/android-udev-rules/template
+++ b/srcpkgs/android-udev-rules/template
@@ -1,7 +1,7 @@
 # Template file for 'android-udev-rules'
 pkgname=android-udev-rules
 version=20200613
-revision=1
+revision=2
 short_desc="Android udev rules list aimed to be the most comprehensive on the net"
 maintainer="Rien Maertens <rien.maertens@posteo.be>"
 license="GPL-3.0-or-later"
@@ -9,6 +9,10 @@ homepage="https://github.com/M0Rf30/android-udev-rules"
 distfiles="https://github.com/M0Rf30/android-udev-rules/archive/${version}.tar.gz"
 checksum=8c7fad8cf97fe106f826f07aa9f033e9aa9ee69422b35dada6d9d2df878b5473
 
+do_patch() {
+	vsed -i -e 's/GROUP="adbusers"/GROUP="users"/' 51-android.rules
+}
+
 do_install() {
 	vinstall 51-android.rules 644 usr/lib/udev/rules.d 51-android.rules
 }
```

Is this good enough? I'll push it out.
I understand your position on not depending on udev rules.

**Comment by Duncaen** [comment]
I think keeping the "adbusers" group is better, but creating it as part of the `android-udev-rules` package.

**Comment by rien** [comment]
@Duncaen when would you not want this package if you install `android-tools`?

**Comment by Duncaen** [comment]
When using the image tools and for me personally I've been using adb with my own rule for my specific device and after rebooting today I noticed error logs from eudev because of it, I'm probably a bit biased against it because it just doesn't work at the moment.

**Comment by rien** [comment]
It was working fine here, but I also had the `adbusers` group already present. I'll create a fix.

**Comment by rien** [comment]
See #23307

**Pull request merged**